src/shared/lib/ListIndexedMap.ts

```diff
@@ -61,4 +61,32 @@ export default class ListIndexedMap<R> {
         this.map[folderKey] = this.map[folderKey] || [];
         return this.map[folderKey];
     }
+}
+
+export class ListIndexedSet {
```

Comment:
makes me nervous because it's library code without tests or documentation.  Also, i'm not sure what "ListIndexed" means.    If I understand correctly this keeping track of a set of keys?

Reply:
yeah. I think it's not an ideal system. I think Im going to go in and refactor this so it accepts an object rather than an amorphous list of strings. ListIndexedMap, which it depends on, is thoroughly tested, I'll add tests for ListIndexedSet specifically